Sources/Sentry/SentryHttpTransport.m

```diff
@@ -184,35 +191,67 @@ - (void)sendCachedEventsAndEnvelopes {
         return;
     }
     
-    dispatch_group_t dispatchGroup = dispatch_group_create();
-    for (SentryFileContents *fileContents in [self.fileManager getAllStoredEventsAndEnvelopes]) {
-        dispatch_group_enter(dispatchGroup);
+    [self sendAllCachedEvents];
+    [self sendAllCachedEnvelopes];
+}
+
+- (void)sendAllCachedEvents {
+    for (SentryFileContents *fileContents in [self.fileManager getAllEventsAndMaybeEnvelopes]) {
+        // The fileContents don't give insights on the event type. We would have
```

Comment:
```suggestion
        // TODO: The fileContents don't give insights on the event type. We would have
```

Reply:
Hm we don't know when we are going to have different categories for event types. If we never get them it doesn't make sense to deserialize the events here as it's a big overhead. Therefore I'm not sure if we should add a TODO here.

Reply:
ok, feel free to decide